### PR TITLE
Refactor Project

### LINKED ISSUE #293: Refactoring and Improving Tests
### Issue To Be Solved
CLI codebase should be improved for better testability. 

### Suggest A Solution
Refactor of the codebase to achieve the following:
- extract all resource loading implementation out of the service layer so we can inject an interface that describes the resource loading implementation and can be injected and mocked during testing,
- refactor project to the "state" and improve the overall structure,
- refactor the "command" and extract the logic into a function which will then be tested as part of the integration testing suite,
- refactor the service API layer so it accepts arguments as their final typed values and not raw string values,
- write more unit testing to cover more codebase,
- refactor integration tests to share common logic and better design the code coverage so we don't have multiple tests testing same functionality,
- support the new automated release process

### PR DESCRIPTION
**Pull request by sideninja**
Closes #293 

## Description
The `project` package was renamed to `state` as it is more accurate description of the package purpose.

The `project.go` was refactored and some code was extracted into `account.go`. 
The newly created state and accounts alongside transactions were moved to the top-level package `flowkit`.

State (project) was also refactored to provide better API for accessing configuration which is an implementation detail and it is now hidden and only accessed through state APIs.

This was also done in preparation for file loader injection.
______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: pkg/flowkit/account.go [hunk 1]
```diff
@@ -121,3 +117,68 @@ func generateEmulatorServiceAccount(sigAlgo crypto.SignatureAlgorithm, hashAlgo
...
 }
+
+// Accounts is a collection of account.
+type Accounts []Account
+
+// Remove an account.
+func (a *Accounts) Remove(name string) error {
+	account := a.ByName(name)
+	if account == nil {
+		return fmt.Errorf("account named %s does not exist in configuration", name)
```
Comment (janezpodhostnik) [inline pkg/flowkit/account.go:128]: This is just personal preference, but I feel Remove operations should be nilpotent.

**Review by janezpodhostnik — APPROVED** [review]
Looks good.

File: internal/keys/generate.go [hunk 2]
```diff
@@ -21,12 +21,13 @@ package keys
 import (
 	"fmt"
 
+	"github.com/onflow/flow-cli/pkg/flowkit"
```
[the remark this answers is not in the document]
Comment (sideninja) [reply internal/keys/generate.go:24]: ohh yeah, I need to setup auto-linting for this not sure why it doesn't work in Goland for me... will fix tnx

File: pkg/flowkit/account.go (hunk 1 above)
Comment (sideninja) [reply pkg/flowkit/account.go:128]: Do you mean idempotent or else please explain not sure what do you mean?

Comment (janezpodhostnik) [reply pkg/flowkit/account.go:128]: Yes sorry, spelling mistake.

I meant that I should be able to call `Remove("x")` multiple times without any problems (errors). Like I said, just a personal preference. Feel free to ignore.

Comment (sideninja) [reply pkg/flowkit/account.go:128]: yeah I kinda agree with you, idempotent functions are always better, will change

**Pull request closed without merging**